src/Microsoft.Framework.Configuration/Internal/ConfigurationFocus.cs

```diff
@@ -72,5 +73,10 @@ public void Set(string key, string value)
         {
             return _root.GetConfigurationSections(_prefix + key);
         }
+
+        public void Reload()
+        {
+            throw new NotImplementedException();
```

Comment:
Hmm interesting. For a sub-section it might be ok to throw, no? Because it doesn't even make sense to call reload on a sub-section. But maybe a message saying that you have to call `Reload` on the main config only? Though, this puts into question whether a "top-level" config might need a slightly different interface that allows reloading, and sub-sections implement a slightly smaller interface...